pkg/api/v1/pod/util.go

```diff
@@ -50,6 +50,47 @@ func FindPort(pod *v1.Pod, svcPort *v1.ServicePort) (int, error) {
 	return 0, fmt.Errorf("no suitable port for manifest: %s", pod.UID)
 }
 
+type ContainerType int
+
+const (
+	InitContainer = 1
+	Container = 2
+	EphemeralContainer = 4
+)
+
+// ContainerVisitorV2 is called with container spec of given container type, and returns true
+// if visiting should continue.
+type ContainerVisitorV2 func(container *v1.Container, containerType ContainerType) (shouldContinue bool)
+
+// VisitContainersV2 invokes the visitor function with a pointer to the container
+// spec of given container type in the given pod spec. If visitor returns false,
+// visiting is short-circuited. VisitContainersV2 returns true if visiting completes,
+// false if visiting was short-circuited.
+func VisitContainersV2(podSpec *v1.PodSpec, visitor ContainerVisitorV2, containerTypes int) bool {
```

Comment:
visitor and containerTypes should be switched so that types proceeds the function, that way it's clear to which container types the following code applies. For example:

```
podutil.VisitContainers(&pod.Spec, podutil.Container|podutil.InitContainer, func(c *Container, t ContainerType) bool {
	doSomething(c)
	return true
})
```